src/client/sandbox/cookie/index.js

```diff
@@ -241,5 +248,16 @@ export default class CookieSandbox extends SandboxBase {
         super.attach(window);
 
         this.windowSync = new WindowSync(window, this, this.messageSandbox);
+
+        for (const { parsedCookies, win, resolve } of this.storedWindowSync) {
```

Comment:
Let's move this code to the 'syncCookieForNotAttachedWindows' function.

Reply:
what about `_processPendingWindowSync`?